### ISSUE TITLE
Port GUI to R shiny

### ISSUE THREAD

**Issue by davidcsterratt** [davidcsterratt/retistruct#56]
It looks like R Shiny could replace our use of the RGtk library. This issue will explore using R Shiny to implement the Retistruct GUI.

**Comment by wikki13** [comment]
Definitely, R shiny is actively maintained also with JS integrations it is very much customisable at UI level. These shiny apps runs in browser so it's more accessible. Challenge comes when you make shiny apps to handle a huge data where you will need at most optimisation.

**Comment by davidcsterratt** [comment]
Thanks for the feedback @wikki13  . I have a student working on this project, and in the New Year we'll need some volunteers to test the new version of Retistruct. Would you be willing to be such a volunteer?

**Comment by wikki13** [comment]
@davidcsterratt, I would like to extent my support being a volunteer.

**Comment by davidcsterratt** [comment]
Thank you!

**Comment by zchou96** [comment]
Hi David! I am also trying to use Retistruct on Mac's R and encountered the same problem with RGtk library. May I also volunteer to test the new version of Retistruct? 
I am working on a retina project and your package would be very helpful. Thank you for keeping this package going!

**Comment by davidcsterratt** [comment]
Things left to fix.

Essential:

- [ ] Demos
- [ ] Bitmap
- [ ] PDF
- [ ] Properties
- [ ]  Prevent crashes - roll-back to previous state

Desirable:

- [ ] DS Magnifier

**Comment by davidcsterratt** [comment]
With the addition of PR #72, the port to Shiny is now working. Retistruct is now back on CRAN, but unfortunately v0.8.0 has a bug (fixed by PR #72), which means that it's not yet possible to install Retistruct via
```
install.packages("retistruct")
```
Once a bug on some Mac M1 setups #74  is fixed, we'll try to get a new version on CRAN.

**Comment by davidcsterratt** [comment]
Closing, now that v0.8.1 is released on CRAN (may take a day or two for Wiindows and Mac binaries to be available).
